src/frame.rs

```diff
@@ -893,6 +898,50 @@ impl Drop for Frame {
     }
 }
 
+pub struct FrameAtomIterator<'a> {
+    frame: &'a Frame,
+    index: usize,
+}
+
+impl<'a> Frame {
+    /// Gets an iterator over atoms
+    ///
+    /// # Example
+    /// ```
+    /// # use chemfiles::{Atom, AtomRef, Frame};
+    /// let mut frame = Frame::new();
+    ///
+    /// frame.add_atom(&Atom::new("O"), [0.0, 0.0, 0.0], None);
+    /// frame.add_atom(&Atom::new("H"), [1.0, 0.0, 0.0], None);
+    ///
+    /// let mut atoms: Vec<AtomRef> = Vec::new();
+    ///
+    /// for atom in frame.iter_atoms() {
+    ///     atoms.push(atom);
+    /// }
+    ///
+    /// assert_eq!(atoms.len(), 2);
+    /// ```
+    pub fn iter_atoms(&'a self) -> FrameAtomIterator<'a> {
+        FrameAtomIterator {
+            frame: self,
+            index: 0,
+        }
+    }
+}
+
+impl<'a> Iterator for FrameAtomIterator<'a> {
+    type Item = AtomRef<'a>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.frame.size() <= self.index {
```

Comment:
`Frame::size` does a FFI call (it calls into C code) everytime, which will prevent some optimizations. You could store the size of the frame inside `FrameAtomIterator` to fix this